lib/commands/-command.js

```diff
@@ -6,7 +6,7 @@ module.exports = Command.extend({
   init() {
     process.env.EMBER_CLI_ELECTRON = true;
 
-    if (this._super && this._super.init) {
+    if (this._super && this._super.init && this._super.init !== this.init) {
```

Comment:
Ideally, I would also prefer to use your suggestions but before making this PR i tried using ES6 syntax (identical to what you're doing @bendemboski) and was met with:
```Function.prototype.apply was called on #<Class>, which is a object and not a function
TypeError: Function.prototype.apply was called on #<Class>, which is a object and not a function```